### ISSUE TITLE
Improve code style for get_umap_layout and get_rasterfairy_layout

### ISSUE THREAD

**Issue by fr1ll** [fr1ll/clip-plot#49]

**Comment by CarloOtano** [comment]
Refactor:
- [ ] get_umap_layout
- [ ] get_rasterfairy_layout

**Comment by CarloOtano** [comment]
Interesting to note that process_single_layout_umap uses cuml_ready. In addition it will use "label" (labels are not used if cuml_ready is used).

**Comment by fr1ll** [comment]
Yes, good point. My feeling is that umap is a small part of the runtime unless you run many umaps, so it's not of much use in `process_single_layout_umap`.

**Comment by CarloOtano** [comment]
Another interesting difference between `process_single_layout_umap` and `process_single_layout_umap` is that a model is saved for
multi layout (pickled) but not for single layout.  

It all has to do how they reuse cached embeddings:

- Single layout
    - Looks for existing layout (umap-<plot_id>.json)
- Multi Layout
    - Loads model (a pickled AlignedUMAP object)
    - Writes each layout

**Comment by CarloOtano** [comment]
I am porting the missing functionality from single layout to multi layout.  I will then deprecate single layout and always use multi layout.

**Comment by CarloOtano** [comment]
> Another interesting difference between `process_single_layout_umap` and `process_single_layout_umap` is that a model is saved for multi layout (pickled) but not for single layout.
> 
> It all has to do how they reuse cached embeddings:
> 
>     * Single layout
>       
>       * Looks for existing layout (umap-<plot_id>.json)
> 
>     * Multi Layout
>       
>       * Loads model (a pickled AlignedUMAP object)
>       * Writes each layout

I need to understand better why the models are saved, loaded and updated on the multi umap layout.   It looks like updating UMAP object is not as simple as updating an AlignedUMAP object.
